numpy/random/mtrand.pyx

```diff
@@ -3933,15 +3968,18 @@ cdef class RandomState:
         cdef int64_t lngood, lnbad, lnsample
 
         # This cast to long is required to ensure that the values are inbounds
-        ongood = <np.ndarray>np.PyArray_FROM_OTF(ngood, np.NPY_LONG, np.NPY_ALIGNED)
-        onbad = <np.ndarray>np.PyArray_FROM_OTF(nbad, np.NPY_LONG, np.NPY_ALIGNED)
-        onsample = <np.ndarray>np.PyArray_FROM_OTF(nsample, np.NPY_LONG, np.NPY_ALIGNED)
+        ongood = <np.ndarray>np.PyArray_FROM_OTF(ngood, np.NPY_INT64, np.NPY_ALIGNED)
```

Comment:
This seems a bit surprising in terms of the behavior on 32bit windows which I think should have a 32but interfer, if I understand the rule correctly.

Reply:
The internal machinery requires int64, and there was a cast to int64 below.  The thing that this _does_ relax, is that previous an int64 input would raise because it cannot be safely cast to long (int32) on windows and 32bits.

(I am happy to change it to intp though!  Just seemed like might as well go to int64 directly.)

Reply:
But that _was_ the point of using `NPY_LONG` here, according to the comment above it. To error out early rather than late.

Reply:
@rkern right, I should remove the comment.  Ensuring an error is why I added the explicit check below.  This ensures that:
1. We still get an error when the value is out of bounds.  (later, but not much)
2. The user can pass a default-integer array.  This would otherwise simply fail otherwise because it is not a safe cast.

So, the relaxation and explicit check seemed unfortunately necessary because I don't want to break default integer array input.  Or am I misunderstanding your comment?